source/grid/grid_generator.cc

```diff
@@ -28,11 +30,1180 @@
+        const unsigned int digit_3 = int(serialnumber[3]) - 48;
+
+        const unsigned int digit_23 = 10 * digit_2 + digit_3;
+
+        const double t = static_cast<double>(digit_23) / 100.0;
+        // 1.0 * digit_23 / 100; // maximum thickness in percentag of the cord
+        double x, y_t;
+
+        std::vector<Point<2>> naca_points;
+
+        if (digit_0 == 0 && digit_1 == 0) // is symmetric
+          {
+            for (unsigned int i = 0; i < number_points; i++)
+              {
+                x = i * 1 / (1.0 * number_points - 1);
+                y_t =
+                  5 * t *
+                  (0.2969 * std::pow(x, 0.5) - 0.126 * x -
+                   0.3516 * std::pow(x, 2) + 0.2843 * std::pow(x, 3) -
+                   0.1036 * std::pow(x, 4)); // half thicknes at a position x
+
+                if (is_upper)
+                  naca_points.push_back({x, y_t});
+                else
+                  naca_points.push_back({x, -y_t});
+              }
+          }
+        else // is asymmetric
+          {
+            for (unsigned int i = 0; i < number_points; i++)
+              {
+                double y_c, dy_c, theta;
+                double m = 1.0 * digit_0 / 100; // max. chamber
+                double p = 1.0 * digit_1 / 10;  // location of max. chamber
+                x        = i * 1 / (1.0 * number_points - 1);
+                if (x <= p)
+                  {
+                    y_c  = m / std::pow(p, 2) * (2 * p * x - std::pow(x, 2));
+                    dy_c = 2 * m / std::pow(p, 2) * (p - x);
+                  }
+                else
+                  {
+                    y_c = m / std::pow(1 - p, 2) *
+                          ((1 - 2 * p) + 2 * p * x - std::pow(x, 2));
+                    dy_c = 2 * m / std::pow(1 - p, 2) * (p - x);
+                  }
+
+                y_t =
+                  5 * t *
+                  (0.2969 * std::pow(x, 0.5) - 0.126 * x -
+                   0.3516 * std::pow(x, 2) + 0.2843 * std::pow(x, 3) -
+                   0.1036 * std::pow(x, 4)); // half thicknes at a position x
+
+                theta = std::atan(dy_c);
+
+                if (is_upper)
+                  {
+                    double x_u = x - y_t * std::sin(theta);
+                    double y_u = y_c + y_t * std::cos(theta);
+                    naca_points.push_back({x_u, y_u});
+                  }
+                else
+                  {
+                    double x_l = x + y_t * std::sin(theta);
+                    double y_l = y_c - y_t * std::cos(theta);
+                    naca_points.push_back({x_l, y_l});
+                  }
+              }
+          }
+        return naca_points;
+      }
+
+
+
+      /**
+       * Set airfoil length (i.e. chord length) to a desired length.
+       * calls function set_airfoil_length() for each vector of the array
+       *
+       * @param array containing upper and lower vector
```

Comment:
I just noticed. The format is:
```suggestion
       * @param input Array containing upper and lower vector.
```
Please adjust the parameters in the other functions similarly.

Reply:
so for each @param we use:
input <Type> <description>. 
E.g., @param input Double desired length. ?

Reply:
Yes.